curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/remember-to-set-the-constructor-property-when-changing-the-prototype.english.md

```diff
@@ -6,8 +6,8 @@ challengeType: 1
 
 ## Description
 <section id='description'>
-There is one crucial side effect of manually setting the <code>prototype</code> to a new object. It erased the <code>constructor</code> property! The code in the previous challenge would print the following for <code>duck</code>:
-<blockquote>console.log(duck.constructor)<br>// prints ‘undefined’ - Oops!</blockquote>
+There is one crucial side effect of manually setting the prototype to a new object. It erases the <code>.constructor</code> property! This property can be used to check which function created the instance object, but since the property has been overwritten, it now gives false results:
```

Comment:
- the dot is not part of the property name
- "object instance" is much more readable (there is no instance type), but even just "object" would do
```suggestion
There is one crucial side effect of manually setting the prototype to a new object. It erases the <code>constructor</code> property! This property can be used to check which function created the object instance, but since the property has been overwritten, it now gives false results:
```

Reply:
The "object instance" doesnt make a lot of sense but then again "instance object" is verbose and I'm sure nobody will get confused with "types" in JS as there are no variable types.
I think using "instance" is pretty clear in this context since we're talking about function constructors. Also, I think the sentence should be like
>which function constructor created the instance

Since, we're talking in terms of OOP and function and function constructors are different entities.

Reply:
FCC is very light on theory (on purpose), so it's normal to use natural language. "instance object" suggests there are other types of objects, which is just confusing. In almost the same way it's more correct to use "constructor function" (it's building an object, not a function). But function is extraneous there anyway.

So I suggest:
> which constructor created the object,
or 
> which constructor function was used to create the object,

Reply:
I agree with "constructor function" but why are we calling it an object instead of instance?
In this [challenge](https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/object-oriented-programming/verify-an-objects-constructor-with-instanceof) we have:
>Anytime a constructor function creates a new object, that object is said to be an instance of its constructor.

Even though "object" isnt incorrect, the "instance" is more technical and is using another OOP term which we've already introduced in the previous challenge.

Reply:
Instance by itself is fine, it's the duo that's meeh.

Reply:
Hah, well we should come to a consensus on this:
 >1. This property can be used to check which constructor function created the object.

>2. This property can be used to check which constructor function created the instance. 

What do you guys think?
cc: @RandellDawson, @Nirajn2311, @cmccormack, @scissorsneedfoodtoo

Reply:
Either is fine by me.

Reply:
@lynxlynxlynx already did a great job reviewing the other points, so I'll just weigh in on this last one. I also think either is fine. If instance was introduced in an earlier challenge, then it should be fine to say "2. This property can be used to check which constructor function created the instance."

The MDN article [Details of the Object Model](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Details_of_the_Object_Model) has an interesting note about "instance" in JavaScript vs. class-based languages. Seems like your use here and in the other challenge you mentioned in line with how they use "instance" throughout the article.

Reply:
I agree with what @scissorsneedfoodtoo says regarding using instance.

Reply:
Awesome, we can go with "instance" then.